### PR TITLE
Bvecvx boost

### PR DESCRIPTION
**Pull request by Darruma**

### CODE REVIEW

File: rewards/boost/boost_utils.py [hunk 1]
```diff
@@ -46,8 +49,12 @@ def calc_boost_balances(
     native = Counter()
     non_native = Counter()
 
+    bvecvx_bals = sett_snapshot(chain, block, addresses.BVECVX, blacklist=True)
```
Comment (mitche50) [inline rewards/boost/boost_utils.py:52]: It doesn't look like we're capturing bveCVX balances in the bveCVX/CVX LP here, just naked holdings of bveCVX?

File: tests/test_utils/test_boost_utils.py [hunk 2]
```diff
@@ -94,3 +101,7 @@ def test_calc_union_addresses():
         "0x0000000000007F150Bd6f54c40A34d7C3d5e9f56",
         "0x0000000000007F150Bd6f54c40A34d7C3d5e9f561",
     }
+
+def test_get_bvecvx_lp_ratio():
+    ratio = get_bvecvx_lp_ratio()
+    assert 0 < ratio < 1
```
Comment (SHAKOTN) [inline tests/test_utils/test_boost_utils.py:107]: Missing whitespace EOF

File: tests/test_utils/test_boost_utils.py [hunk 3]
```diff
@@ -1,20 +1,26 @@
...
     NFT_SNAPSHOT_DATA,
 )
 
+@pytest.fixture
+def mock_fns(mocker):
+    mocker.patch(
+        "rewards.boost.boost_utils.claims_snapshot",
+        return_value=({}),
+    )
+def test_calc_boost_balances(chain, mock_snapshots, mock_fns):
```
Comment (SHAKOTN) [inline tests/test_utils/test_boost_utils.py:21]: Missing 2 whitespaces before `test_calc_boost_balances` function

File: tests/test_utils/test_boost_utils.py [hunk 4]
```diff
@@ -1,20 +1,26 @@
...
+    get_bvecvx_lp_ratio,
 )
 from tests.conftest import (
     CHAIN_CLAIMS_SNAPSHOT_DATA,
     CHAIN_SETT_SNAPSHOT_DATA,
     NFT_SNAPSHOT_DATA,
 )
 
+@pytest.fixture
+def mock_fns(mocker):
```
Comment (SHAKOTN) [inline tests/test_utils/test_boost_utils.py:16]: Can it be named more verbose, like `mock_claims_snapshot` or something?

File: rewards/classes/Boost.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+from typing import Dict
+from decimal import Decimal
+
+
+@dataclass
+class BoostBalances:
+    native: Dict[str, Decimal]
+    non_native: Dict[str, Decimal]
+    bvecvx: Dict[str, Decimal]
+    nfts: Dict[str, Decimal]
```
Comment (SHAKOTN) [inline rewards/classes/Boost.py:11]: MIssing whitespace eof

File: rewards/boost/calc_boost.py [hunk 6]
```diff
@@ -8,21 +8,23 @@
...
     :param native_setts: native balances
     :param non_native_setts: non native balances
     """
-    native_balance = native_setts.get(address, 0)
-    non_native_balance = non_native_setts.get(address, 0)
+    native_balance = boost_bals.native.get(address, 0)
+    non_native_balance = boost_bals.non_native.get(address, 0)
+    bvecvx_balance = boost_bals.bvecvx.get(address, 0)
+    if bvecvx_balance > 0 and native_balance > 0:
+        native_balance = native_balance + min(0.5 * bvecvx_balance, 0.5 * native_balance)
```
Comment (SHAKOTN) [inline rewards/boost/calc_boost.py:27]: Maybe we should move these `0.5` numbers to constants?

File: rewards/classes/Boost.py [hunk 7]
```diff
@@ -0,0 +1,11 @@
+from dataclasses import dataclass
+from typing import Dict
+from decimal import Decimal
+
+
+@dataclass
+class BoostBalances:
+    native: Dict[str, Decimal]
+    non_native: Dict[str, Decimal]
+    bvecvx: Dict[str, Decimal]
```
Comment (SHAKOTN) [inline rewards/classes/Boost.py:10]: Are `bvecvx` and `nfts` balance optional? If yes, let's make them optional here, and not pass empty dicts ther

**Review by SHAKOTN — COMMENTED** [review]
Looks good, mostly style comments

File: rewards/boost/boost_utils.py [hunk 8]
```diff
@@ -1,17 +1,24 @@
...
 from rewards.snapshot.token_snapshot import token_snapshot_usd
-
+from rewards.classes.Boost import BoostBalances
 console = Console()
 
 
+def get_bvecvx_lp_ratio() -> float:
+    bvecvx_pool = make_contract(addresses.BVECVX_CVX_LP, Abi.Stableswap, Network.Ethereum)
+    balances = bvecvx_pool.get_balances().call()
+    return balances[1]/ sum(balances) # balances contains [cvx, bcvx]
```
Comment (btcookies) [inline rewards/boost/boost_utils.py:20]: should be [cvx, bvecvx]?

**Review by SHAKOTN — APPROVED** [review]

File: rewards/boost/boost_utils.py [hunk 9]
```diff
@@ -45,9 +50,9 @@ def calc_boost_balances(
 
     native = Counter()
     non_native = Counter()
-
     console.log(f"\n === Taking claims snapshot on {chain} === \n")
     native_claimable, non_native_claimable = claims_snapshot_usd(chain, block)
+    bvecvx_claimable = claims_snapshot(chain, block).get(addresses.BVECVX)
```
Comment (btcookies) [inline rewards/boost/boost_utils.py:55]: why do we do this again in if block?

**Review by mitche50 — APPROVED** [review]

**Review by mitche50 — DISMISSED** [review]

**Review by mitche50 — APPROVED** [review]

**Pull request merged**